deepchem/feat/material_featurizers/lcnn_featurizer.py

```diff
@@ -0,0 +1,656 @@
+    self.formula = defaultdict(int)
+    for s in sitetypes:
+      self.formula[s] += 1
+    self.permutations = permutations
+    self.env2config = env2config
+    self.cutoff = cutoff
+    # Set up site environment matcher
+    self.tol = tol
+    # Graphical option
+    self.Grtol = Grtol
+    self.Gatol = Gatol
+    # tolerance for grouping nodes
+    self.grtol = 1e-3
+    # determine minimum distance between sitetypes.
+    # This is used to determine the existence of an edge
+    dists = squareform(pdist(pos))
+    mindists = defaultdict(list)
+    for i, row in enumerate(dists):
+      row_dists = defaultdict(list)
+      for j in range(0, len(sitetypes)):
+        if i == j:
+          continue
+        # Sort by bond
+        row_dists[frozenset((sitetypes[i], sitetypes[j]))].append(dists[i, j])
+      for pair in row_dists:
+        mindists[pair].append(np.min(row_dists[pair]))
+    # You want to maximize this in order to make sure every node gets an edge
+    self.mindists = {}
+    for pair in mindists:
+      self.mindists[pair] = np.max(mindists[pair])
+    # construct graph
+    self.G = self._ConstructGraph(pos, sitetypes)
+    # matcher options
+    self._nm = iso.categorical_node_match('n', '')
+    self._em = iso.numerical_edge_match('d', 0, rtol, 0)
+
+  def _ConstructGraph(self, pos, sitetypes):
+    """
+    Returns local environment graph using networkx and
+    tolerance specified.
+
+    parameters
+    ----------
+    pos: ns x 3. coordinates of positions. ns is the number of sites.
+    sitetypes: ns. sitetype for each site
+
+    Returns
+    ------
+    networkx graph used for matching site positions in
+    datum.
+    """
+    try:
+        import networkx as nx
+    except:
+        raise ImportError("This class requires networkx to be installed.")
+
+    try:
+        from scipy.spatial.distance import cdist, pdist
+    except:
+        raise ImportError("This class requires scipy to be installed.")
+
+    # construct graph
+    G = nx.Graph()
+    dists = cdist([[0, 0, 0]], pos - np.mean(pos, 0))[0]
+    sdists = np.sort(dists)
+    # https://stackoverflow.com/questions/37847053/uniquify-an-array-list-with-a-tolerance-in-python-uniquetol-equivalent
+    uniquedists = sdists[
+        ~(np.triu(np.abs(sdists[:, None] - sdists) <= self.grtol, 1)).any(0)]
+    orderfromcenter = np.digitize(dists, uniquedists)
+    # Add nodes
+    for i, o in enumerate(orderfromcenter):
+      G.add_node(i, n=str(o) + sitetypes[i])
+    # Add edge. distance is edge attribute
+    dists = pdist(pos)
+    n = 0
+    for i in range(len(sitetypes)):
+      for j in range(i + 1, len(sitetypes)):
+        if dists[n] < self.mindists[frozenset((sitetypes[i], sitetypes[j]))] or\
+            (abs(self.mindists[frozenset((sitetypes[i], sitetypes[j]))] - dists[n]) <= self.Gatol + self.Grtol * abs(dists[n])):
+          G.add_edge(i, j, d=dists[n])
+        n += 1
+    return G
+
+  def GetMapping(self, env, path=None):
```

Comment:
For DeepChem method names, we usually follow the snake case style. So this would be `def get_mapping(...`